Manage Logon Hour via Windows PowerShell

```diff
@@ -0,0 +1,345 @@
+---
+post_title: Manage Logon Hour via Windows PowerShell
+username: farisnt@gmail.com
+Catagories: PowerShell, Active Directory
+Summary: How to change users Logon Hours using PowerShell
+---
+
+
+**Q:** How can I set AD domain user Logon Hour via Windows PowerShell ?
+
+**A:** Most of the Active Directory object properties can be easily read or modified by using the `Get-` or `Set-` with the property name and its value, but when it comes to **LogonHours**, it not that easy. There are multiple steps and challenges required to build the value, and in this tutorial, you will learn all the tips and tricks needed to set the **LogonHours** hours via Windows PowerShell. So get a cup of coffee, and let's start.
+
+## Logon Hours, Understanding the basics.
+
+First, let's start with the GUI of the Logon Hours property by opening any AD user account property —> **Account** Tab —> click on **Logon Hours.** The default value is all blue, which represents logon permitted. If there is any white block, it represents a logon denied on that selected time/day slot.
+
+In the example below, the user cannot log in on Sunday or Saturday all day but can log in from Monday through Friday. 
+
+![Manage%20Logon%20Hour%20via%20Windows%20PowerShell%20c28909d216f44708aad42cfa6706b403/Untitled.png](Manage%20Logon%20Hour%20via%20Windows%20PowerShell%20c28909d216f44708aad42cfa6706b403/Untitled.png)
+
+Logon Hours GUI
+
+Make to enable the Advanced Features in Active Directory Management Console to see the Attribute Editor for the users, open the Active Directory console, click on **View** Menu and select **Advanced Features**. 
+
+Start by selecting any user —> **Attribute Editor** tab. From the attributes, scroll to **LogonHours** and double click on it. Change the **Value Format** to **Binary.** 
+
+![Manage%20Logon%20Hour%20via%20Windows%20PowerShell%20c28909d216f44708aad42cfa6706b403/Untitled%201.png](Manage%20Logon%20Hour%20via%20Windows%20PowerShell%20c28909d216f44708aad42cfa6706b403/Untitled%201.png)
+
+Logon Hour in attribute
+
+This is what represent the logon hours, but this is undoubtedly not a readable value and its not clear what these value means.
+
+## Understanding LogonHour Property
+
+In the **LogonHour**, each day of the week represented with 3 Byte (Three sets of **11111111** or **00000000**), and each 1 bit of the byte represent 1 hour
```

Comment:
This is unclear. I think what you mean is that:
...each day of the week is represented by a three-byte (24-bit) field, with each bit representing 1 specific hour.

Reply:
Yes, exactly, I took this sentence and place it in the post.